src/client/components/Reservations.js

```diff
@@ -1,5 +1,119 @@
-import React from "react";
+import React, { useState, useContext } from "react";
+import Header from "./Header";
+import Footer from "./Footer";
+import DataContext from "./DataContext";
 
 export default function Reservations() {
-  return <h2>Reservations</h2>;
+  const { upcomingMeals, reservations, loading, error } =
+    useContext(DataContext);
+  if (loading) return <h3>Loading...</h3>;
+  if (error) return <p>{error}</p>;
+
+  const [inputState, setInputstate] = useState(false);
+  const [name, setName] = useState("");
+  const [phoneNumber, setPhoneNumber] = useState("");
+  const [numberOfGuests, setNumberOfGuests] = useState("");
+  const [email, setEmail] = useState("");
+  const [mealId, setMealId] = useState("");
+
+  function newReservation() {
```

Comment:
Also feel free to extract this function to it's own file. I don't need to read the details of this function to understand the component and vice versa. This just gives the readers more details which they might not be looking for at that time.